src/services/VirtualFileSystemService.spec.ts

```diff
@@ -1,532 +1,423 @@
-            },
-          },
-        },
-        srcSL: {
-          type: MetadataType.symlink,
-          target: 'src',
-        },
-      },
-    };
-
-    service.setFileSystem(structure);
-  });
+    const container = createTestContainer({ uri: '.' });
+    virtualFileSystemService = container.virtualFileSystemService;
 
-  afterEach(async () => {
-    service.clearFileSystem();
+    virtualFileSystemService.setFileSystem({
+      '/mockFolder/mockSubFolder/mockSubFolderFile.txt': '',
+      '/mockFolder/mockFileSLbroken.ln': '',
+      '/mockFolder/mockFileToRewrite.ts': '',
+      '/mockFolder/mockFile.ts': '',
+      '/.keep': '',
+      '/index.ts': '',
+    });
   });
 
   describe('#exists', () => {
     it('returns true if the file exists', async () => {
-      const result = await service.exists('/src/index.ts');
-      expect(result).toBe(true);
-    });
-
-    it('returns true if the directory is a root directory', async () => {
-      const result = await service.exists('/');
-      expect(result).toBe(true);
-    });
-
-    it('returns true if the directory exists on absolute path', async () => {
-      const result = await service.exists('/src');
-      expect(result).toBe(true);
-    });
-
-    it('returns true if the directory exists on relative path', async () => {
-      service.setFileSystem({
-        type: MetadataType.dir,
-        children: {
-          src: {
-            type: MetadataType.dir,
-            children: {},
-          },
-        },
-      });
-
-      const result = await service.exists('src');
-      expect(result).toBe(true);
-    });
-
-    it('returns true if the symbolic link exists', async () => {
-      const result = await service.exists('/src/indexSL.ts');
-      expect(result).toBe(true);
+      const mockFilePath = path.resolve('/mockFolder/mockFile.ts');
+      const result = await virtualFileSystemService.exists(mockFilePath);
+      expect(result).toEqual(true);
     });
 
-    it('returns false if the broken symbolic link exists', async () => {
-      const result = await service.exists('/src/indexSLbroken.ts');
-      expect(result).toBe(false);
+    it('returns true if the directory exists', async () => {
+      const mockFolderPath = path.resolve('/mockFolder');
+      const result = await virtualFileSystemService.exists(mockFolderPath);
+      expect(result).toEqual(true);
     });
 
     it("returns false if the file doesn't exists", async () => {
-      const result = await service.exists('/non/existing/file.ts');
-      expect(result).toBe(false);
+      const mockFolderPath = path.resolve('/notExistingMockFolder');
+      const result = await virtualFileSystemService.exists(mockFolderPath);
+      expect(result).toEqual(false);
     });
   });
 
   describe('#readDirectory', () => {
     it('returns array of files after calling readDirectory()', async () => {
-      const result = await service.readDirectory('/src');
-      expect(result.length).toEqual(5);
```

Comment:
What about preserving this expectation as well in order to make sure that no more filenames is returned?